src/filter/FeatureFilter.ts

```diff
@@ -3,7 +3,7 @@
 
 export interface IFeatureFilter {
     name: string; // e.g. Microsoft.TimeWindow
-    evaluate(context: IFeatureFilterEvaluationContext, appContext?: unknown): Promise<boolean> | boolean;

```

Comment:
This interface should just return Promise<boolean>. The calling code is forced to use await anyway since this function might return a Promise. If await is used on a non-promise, it simply wraps it in an immediately resolved Promise anyway. 

I believe the code can just stay "return true" and things will work fine.